students/brandon_nguyen/session04/mailroom2.py

```diff
@@ -0,0 +1,143 @@
+#building switch case menu with dictionary
+
+
+#The script should prompt the user (you) to choose from a menu of 3 actions: “Send a Thank You”, “Create a Report” or “quit”.
+
+#think about a main menu function that can be called any where.
+def show_menu(promptxt, select_dict):
+    while True:
+        menu_selected = input(promptxt)
+        if select_dict[menu_selected]() == "Exit Menu":
+            break
+
+def send_ty():
+    show_menu(subMenuPrompt, sub_menu_dict)
+
+def create_rpt():
+    """
+    This function is to print the report of donor.  
+    """
+    #Currently need more work on the format as 0.0 problem not yet solved.
+    print()
+    newList = sort_sum() #creating a new list with sorted total in reverse already for easy printing.
+    header_string = "Donor Name              |  Total Given | Num Gifts | Average Gift"
+    line = '-'*len(header_string)
+    print(header_string)
+    print(line)
+    for donor, total, num, avg in newList:
+        print(" {:<18}     $  {:>12.2f} {:>6} {:>6} {:>8.2f}".format(donor,total[0],num[0],"$",avg[0]))
+    print("\n\n")
+      
+def sort_sum():
+    #creating a new list with computed value for easy printing.
+    """
+    This function returned a sorted list of by on order amount3
+    """
+    dblist = []
+    dbSum = []
+    #quick way to convert back to list to reuse existing code.
+    for i,j in donor_db.items():
+        dblist.append((i,j))
+    for name, donation in dblist:
+        dbSum.append((name,[round(sum(donation),2)],[len(donation)],[round(sum(donation)/len(donation),2)]),)
+    return sorted(dbSum, key=lambda donor: donor[1], reverse=True)
+    
+def list_donors():
+    """
+    Basic Name of donor return in Order by First Name
+    """
+    lst = []
+    print()
+    for person in donor_db:
+        lst.append(person)
+    lst.sort()
+    for name in lst: print(name)
+    print()
+
+def update_donation():
+        input_person = input("\nPlease enter donor full names: " )
+        input_donation = float(input("Please enter donation amount for " + input_person + ":>> "))
+        #existing people add to donation
+        if input_person in donor_db:
+            new_Donation = donor_db.get(input_person)
+            new_Donation.append(input_donation)
+            donor_db[input_person] = new_Donation
+        else:
+            donor_db.update( {input_person :[input_donation]} ) 
+        print()
+        print(email_template(input_person,input_donation))
+
+def email_template(name, newAmout):
+    e_frmt = {
+                  0:'Dear',
+                  1:'Thank you for your kind donation of ',
+                  2:'It will be put to very good use',
+                  3:'Sincerely, ',
+                  4:'-The Team'
+                  }
+    #Not the best way to do this.
+    txt = ("\n{} {},\n\n {:>45}${:.2f}.\n\n {: >40}.\n\n{:>40}\n"
+         "{:>42}".format(e_frmt.get(0),name,e_frmt.get(1), newAmout,e_frmt.get(2),e_frmt.get(3),e_frmt.get(4)))
```

Comment:
This looks unnecessarily concocted--I think I heard you saying that already in class... :)

I think the intention of the `dict` use requirement with `.format(...)` was to exercise the keyword arguments with a dict expansion. No need to change this for that now.